includes/class-wc-youcanpay-api.php

```diff
@@ -4,247 +4,264 @@
+                    print_r($token, true),
+                    __('There was a problem connecting to the YouCan Pay API endpoint.', 'youcan-pay')
+                );
+            }
+
+            $response = [
+                'id'       => $token->getId(),
+                'redirect' => [
+                    'url' => $token->getPaymentURL($post_data['locale']),
+                ],
+            ];
+        } catch (Throwable $e) {
+            WC_YouCanPay_Logger::alert('throwable at request exists into wc youcan pay api', [
+                'exception.message' => $e->getMessage(),
+            ]);
+        }
+
+        return json_decode(json_encode($response));
+    }
+
+    /**
+     * @return Transaction|null
+     */
+    public static function get_transaction($transaction_id)
+    {
+        try {
+            if (self::is_test_mode()) {
+                YouCanPay::setIsSandboxMode(true);
+            }
+            $py = YouCanPay::instance()->useKeys(
+                self::get_private_key(),
+                self::get_public_key()
+            );
+
+            return $py->transaction->get($transaction_id);
+        } catch (Throwable $e) {
+            WC_YouCanPay_Logger::alert('throwable at get transaction exists into wc youcan pay api', [
+                'exception.message' => $e->getMessage(),
+            ]);
+        }
+
+        return null;
+    }
+
+    /**
+     * @param $order WC_Order
+     * @param $total
+     * @param $currency
+     * @param $return_url
+     *
+     * @return Token|null
+     */
+    public static function create_token($order, $total, $currency, $return_url)
+    {
+        try {
+            $amount = WC_YouCanPay_Helper::get_youcanpay_amount($total, $currency);
+
+            if (self::is_test_mode()) {
+                YouCanPay::setIsSandboxMode(true);
+            }
+
+            $py = YouCanPay::instance()->useKeys(
+                self::get_private_key(),
+                self::get_public_key()
+            );
+
+            return $py->token->create(
+                $order->get_id(),
+                $amount,
+                strtoupper($currency),
+                self::get_the_user_ip(),
+                $return_url,
+                $return_url
+            );
+        } catch (Throwable $e) {
+            WC_YouCanPay_Logger::alert('throwable at create token exists into wc youcan pay api', [
+                'exception.message' => $e->getMessage(),
+            ]);
+        }
+
+        return null;
+    }
+
+    public static function get_the_user_ip()
+    {
+        if (!empty($_SERVER['HTTP_CLIENT_IP'])) {
+            //check ip from share internet
+            $ip = $_SERVER['HTTP_CLIENT_IP'];
+        } elseif (!empty($_SERVER['HTTP_X_FORWARDED_FOR'])) {
+            //to check ip is pass from proxy
+            $ip = $_SERVER['HTTP_X_FORWARDED_FOR'];
+        } else {
+            $ip = $_SERVER['REMOTE_ADDR'];
+        }
+
+        return $ip;
+    }
```

Comment:
```
    public static function get_the_user_ip()
    {
        if (!empty($_SERVER['HTTP_CLIENT_IP'])) {
            //check ip from share internet
            return  $_SERVER['HTTP_CLIENT_IP'];
        }
        
        if (!empty($_SERVER['HTTP_X_FORWARDED_FOR'])) {
            //to check ip is pass from proxy
            return $_SERVER['HTTP_X_FORWARDED_FOR'];
        }

        return $_SERVER['REMOTE_ADDR'];
    }
```

Reply:
we can refactor this 🙏

Reply:
https://github.com/NextmediaMa/youcan-pay-for-woocommerce/pull/12/commits/3583c00247402892d45f5c0af08d6827e0a5e5b5